### DISCUSSION TITLE
How do I run the SIONNA 5G-LDPC codes for Code-rate = k/n = 4632 / 16384 = 0.282714844 ?

### DISCUSSION THREAD

**Discussion by yossefyossef** [NVlabs/sionna#110] [Q&A]
Hi all,

Can someone please help me, how to run the SIONNA for the following code-rate?

Code-rate = k/n = 4632/16384 = 0.282714844
k = 4632 = number of information bits per code word.
n = 16384 = number of bits per coded word.

I use QPSK modulation, i.e. bits_per_symbol = 2

I got the following error message: 
“Repetition coding is currently not supported”.

Can someone please help me?

Thanking in advance, and best regards,
Yossef Yossef
Email: yossiy@gilat.com

Note:
I use the code from the following link:
https://nvlabs.github.io/sionna/examples/5G_Channel_Coding_Polar_vs_LDPC_Codes.html

**Comment by SebastianCa** [comment]
Hi Yossef,

Sionna does currently not support your exact code parameters (as repetition coding in the rate matcher is required for the resulting *basegraph*).

You have multiple options depending on your exact problem formulation:
- Increase the code rate (increase k or decrease n) such that r>1/3
- Decrease the length (e.g., encode into two codewords each with k/2 and n/2) such that "basegraph2" is used
- Implement your own repetition coding scheme (which should be only a few lines of code)

**Reply by rustyolford** [reply]
Hi Yossef.  Do you know if support for lower code rates will be added in the next release?  To me, it seems like an important feature to be able to simulate lower MCS values required by the 3GPP spec.

Thanks,
Russell
